src/lib.rs

```diff
@@ -37,11 +37,26 @@ mod errors {
                 description("unrecognizable charset")
                 display("uchardet was unable to recognize a charset")
             }
-            DataHandlingError {
-                description("data handling error")
-                display("uchardet failed to handle the supplied data (out of memory?)")
+            OutOfMemory {
+                description("out of memory error")
+                display("uchardet ran out of memory")
+            }
+            Other(int: i32) {
+                description("unknown error")
+                display("uchardet returned unknown error {}", int)
+            }
+        }
+    }
+
+    impl ErrorKind {
+        pub fn from_nsresult(nsresult: ::ffi::nsresult) -> ErrorKind {
```

Comment:
I put it in a module so I could `#[allow(missing_docs)]` as it appears that you're unable to document the generated code (and I don't see another way to set that attribute). I'm currently trying to write up a PR for `error-chain` so the generated code has `#[allow(missing_docs)]`.

It's no problem at all; Rather it's really educational for me to have somebody more experienced guide me.